### PR TITLE
State Machine Manager

### PR DESCRIPTION
**Pull request by dharjeezy**
A pallet for freezing and unfreezing state machines via the ISMP Host

### CODE REVIEW

File: parachain/modules/ismp/pallet/src/host.rs [hunk 1]
```diff
@@ -102,10 +102,10 @@ impl<T: Config> IsmpHost for Host<T> {
         <T::TimeProvider as UnixTime>::now()
     }
 
-    fn is_state_machine_frozen(&self, machine: StateMachineHeight) -> Result<(), Error> {
-        if let Some(frozen_height) = FrozenHeights::<T>::get(machine.id) {
-            if machine.height >= frozen_height {
-                Err(Error::FrozenStateMachine { height: machine })?
+    fn is_state_machine_frozen(&self, machine: StateMachineId) -> Result<(), Error> {
+        if let Some(frozen) = FrozenStateMachine::<T>::get(machine) {
+            if frozen == true {
```
Comment (Wizdave97) [inline parachain/modules/ismp/pallet/src/host.rs:107]:
```suggestion
            if frozen {
```

File: parachain/modules/ismp/testsuite/src/mocks.rs [hunk 2]
```diff
@@ -162,15 +162,15 @@ impl IsmpHost for Host {
...
 
-    fn is_state_machine_frozen(&self, machine: StateMachineHeight) -> Result<(), Error> {
+    fn is_state_machine_frozen(&self, machine: StateMachineId) -> Result<(), Error> {
         let val = self
             .frozen_state_machines
             .borrow()
-            .get(&machine.id)
-            .map(|frozen_height| machine.height >= frozen_height.height)
+            .get(&machine)
+            .map(|frozen| frozen == true)
```
Comment (Wizdave97) [inline parachain/modules/ismp/testsuite/src/mocks.rs:170]:
```suggestion
```

File: parachain/modules/state-machine-freezer/Cargo.toml [hunk 3]
```diff
@@ -0,0 +1,32 @@
+[package]
+name = "pallet-state-machine-freezer"
```
Comment (Wizdave97) [inline parachain/modules/state-machine-freezer/Cargo.toml:2]:
```suggestion
name = "state-machine-manager"
```

Comment (Wizdave97) [inline parachain/modules/state-machine-freezer/Cargo.toml:2]:
```suggestion
name = "state-machine-manager"
```

**Review by Wizdave97 — CHANGES_REQUESTED** [review]

File: Cargo.toml [hunk 4]
```diff
@@ -37,7 +37,7 @@ members = [
 
      # evm stuff
       "evm/integration-tests",
-      "evm/abi", "parachain/modules/relayer-fees",
+      "evm/abi", "parachain/modules/relayer-fees", "parachain/modules/state-machine-manager",
```
Comment (seunlanlege) [inline Cargo.toml:40]: Should all be on new lines

File: parachain/runtimes/gargantua/src/lib.rs [hunk 5]
```diff
@@ -557,6 +557,7 @@ construct_runtime!(
         IsmpDemo: ismp_demo = 42,
         RelayerFees: pallet_relayer_fees = 43,
         IsmpPolygonPos: ismp_polygon_pos::pallet = 44,
+        StateMachineManager: state_machine_manager = 45,
```
Comment (Wizdave97) [inline parachain/runtimes/gargantua/src/lib.rs:560]: The pallet config is not implemented for the runtime

File: parachain/modules/state-machine-manager/Cargo.toml [hunk 6]
```diff
@@ -0,0 +1,32 @@
+[package]
+name = "state-machine-manager"
+version = "0.1.0"
+edition = "2021"
```
Comment (seunlanlege) [inline parachain/modules/state-machine-manager/Cargo.toml:4]: author and description

**Review by Wizdave97 — APPROVED** [review]

**Review by seunlanlege — APPROVED** [review]

**Pull request merged**